### PR TITLE
lomiri.*: OTA-5

### PR DESCRIPTION
**Pull request by OPNA2608**
## Description of changes

Updates that were pushed for OTA-5: https://ubports.com/en/blog/ubports-news-1/post/ubuntu-touch-ota-5-focal-release-3933

- lomiri-clock-app: 4.0.3 -> 4.0.4
- lomiri-system-settings-security-privacy: Drop
  Absorbed into new LSS release.
- lomiri-system-settings: 1.1.0 -> 1.2.0, nixfmt
- nixos/ayatana-indicators: new systemd target, nixfmt
  Lomiri now uses `lomiri-indicators.target` instead of `ayatana-indicators.target`, because it has Ayatana-like indicators that should only be brought up under Lomiri. Services specify the target they would like to be brought up with.
  Short-term solution: Duplicate `ayatana-indicators.target` with the new name. Long-term, the indicators should get passthru values indicating which target they want to follow.
- lomiri: 0.2.1 -> 0.3.0, nixfmt
- lomiri-session: 0.2 -> 0.3, nixfmt
- tests/lomiri: Fix polkit agent test, nixfmt
  We're starting to time out waiting for OCR on the polkit notification. Just assume that it's there, send password, and expect success.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/desktops/lomiri/data/lomiri-session/default.nix [hunk 1]
```diff
@@ -1,153 +1,40 @@
...
-    substituteInPlace systemd/lomiri.service.in \
-      --replace-fail '/usr/bin/lomiri-session' '@CMAKE_INSTALL_FULL_BINDIR@/lomiri-session' \
-      --replace-fail '/usr/bin/dbus-update-activation-environment' '@DUAE_BIN@'
-
-    substituteInPlace lomiri-session \
+    substituteInPlace lomiri-session.in \
       --replace-fail '/usr/libexec/Xwayland.lomiri' '${lib.getBin lomiri}/libexec/Xwayland.lomiri'
+
+    substituteInPlace systemd/CMakeLists.txt \
+      --replace-fail 'pkg_get_variable(SYSTEMD_USER_DIR systemd systemduserunitdir)' 'pkg_get_variable(SYSTEMD_USER_DIR systemd systemduserunitdir DEFINE_VARIABLES prefix=''${CMAKE_INSTALL_PREFIX})'
```
Comment (wineee) [inline pkgs/desktops/lomiri/data/lomiri-session/default.nix:37]: Set environment variables for pkgconfig should be the simpler way

https://github.com/NixOS/nixpkgs/blob/a1521bc2d16d34067ee45aac75387eaa63f638c0/pkgs/desktops/deepin/core/dde-shell/default.nix#L74

File: pkgs/desktops/lomiri/default.nix [hunk 2]
```diff
@@ -1,65 +1,68 @@
...
-    inherit (self) callPackage;
-  in {
-    #### Core Apps
-    lomiri = callPackage ./applications/lomiri { };
-    lomiri-calculator-app = callPackage ./applications/lomiri-calculator-app { };
-    lomiri-camera-app = callPackage ./applications/lomiri-camera-app { };
-    lomiri-clock-app = callPackage ./applications/lomiri-clock-app { };
-    lomiri-filemanager-app = callPackage ./applications/lomiri-filemanager-app { };
-    lomiri-system-settings-unwrapped = callPackage ./applications/lomiri-system-settings { };
-    lomiri-system-settings-security-privacy = callPackage ./applications/lomiri-system-settings/plugins/lomiri-system-settings-security-privacy.nix { };
```
Comment (wineee) [inline pkgs/desktops/lomiri/default.nix:17]: Need aliase for package removed

**Review by wineee — CHANGES_REQUESTED** [review]

File: pkgs/desktops/lomiri/default.nix (hunk 2 above)
Comment (OPNA2608) [reply pkgs/desktops/lomiri/default.nix:17]: Added an alias for `lomiri-system-settings-security-privacy` -> `lomiri-system-settings-unwrapped`.

File: pkgs/desktops/lomiri/data/lomiri-session/default.nix (hunk 1 above)
Comment (OPNA2608) [reply pkgs/desktops/lomiri/data/lomiri-session/default.nix:37]: That's simpler, but it circumvents the whole pkg-config process abit too much for my taste tbh.

**Review by wineee — APPROVED** [review]

**Pull request merged**